src/middlewared/middlewared/plugins/rsync.py

```diff
@@ -248,3 +249,53 @@ def copy(self, job, rcopy):
                 password_file.close()
 
         job.set_progress(100, 'Rsync copy job successfully completed')
+
+
+class RsyncdService(SystemServiceService):
+
+    class Config:
+        service = "rsync"
+        service_model = 'rsyncd'
+        datastore_prefix = "rsyncd_"
+
+    @accepts(Dict(
+        'rsyncd_update',
+        Int('port', validators=[Range(min=1, max=65535)]),
+        Str('auxiliary')
+    ))
+    async def update(self, data):
+        old = await self.config()
+
+        new = old.copy()
+        new.update(data)
+
+        await self._update_service(old, new)
+
+        return new
+
+
+class RsyncdModService(CRUDService):
```

Comment:
You'll probably also need `do_delete` method here.

Reply:
Agree. Not needed now but it will be needed in the future.